x/cellarfees/keeper/abci_test.go

```diff
@@ -134,3 +146,185 @@ func (suite *KeeperTestSuite) TestBeginBlockerEmissionGreaterThanRewardSupply()
 
 	require.NotPanics(func() { cellarfeesKeeper.BeginBlocker(ctx) })
 }
+
+func (suite *KeeperTestSuite) TestAuctionBeginWithSufficientFunds() {
+	ctx, cellarfeesKeeper := suite.ctx, suite.cellarfeesKeeper
+	cellarfeesKeeper.SetFeeAccrualCounters(ctx, cellarfeesTypes.DefaultFeeAccrualCounters())
+	suite.SetupHooksTests(ctx, cellarfeesKeeper)
+
+	require := suite.Require()
+	params := cellarfeesTypes.DefaultParams()
+	params.AuctionInterval = 1
+	cellarfeesKeeper.SetParams(ctx, params)
+	cellarfeesKeeper.SetLastRewardSupplyPeak(ctx, sdk.NewInt(1000000))
+
+	hooks := Hooks{k: cellarfeesKeeper}
+	event := gravitytypes.SendToCosmosEvent{
+		CosmosReceiver: feesAccount.GetAddress().String(),
+		Amount:         sdk.NewInt(2),
+		EthereumSender: "0x0000000000000000000000000000000000000000",
+		TokenContract:  "0x1111111111111111111111111111111111111111",
+	}
+	cellarID := common.HexToAddress(event.EthereumSender)
+	cellarfeesKeeper.SetFeeAccrualCounters(ctx, cellarfeesTypes.FeeAccrualCounters{
+		Counters: []cellarfeesTypes.FeeAccrualCounter{
+			{
+				Denom: gravityFeeDenom,
+				Count: 2,
+			},
+		},
+	})
+	expectedCounters := cellarfeesTypes.FeeAccrualCounters{
+		Counters: []cellarfeesTypes.FeeAccrualCounter{
+			{
+				Denom: gravityFeeDenom,
+				Count: 0,
+			},
+		},
+	}
+
+	suite.corkKeeper.EXPECT().HasCellarID(ctx, cellarID).Return(true)
+	suite.gravityKeeper.EXPECT().ERC20ToDenomLookup(ctx, common.HexToAddress(event.TokenContract)).Return(false, gravityFeeDenom)
+	suite.accountKeeper.EXPECT().GetModuleAccount(ctx, cellarfeesTypes.ModuleName).Return(feesAccount)
+
+	require.NotPanics(func() { hooks.AfterSendToCosmosEvent(ctx, event) })
+
+	// mocks
+	suite.accountKeeper.EXPECT().GetModuleAccount(ctx, cellarfeesTypes.ModuleName).Return(feesAccount)
+
+	suite.bankKeeper.EXPECT().GetBalance(ctx, feesAccount.GetAddress(), appParams.BaseCoinUnit).Return(sdk.NewCoin(gravityFeeDenom, sdk.NewInt(0)))
+
+	suite.bankKeeper.EXPECT().GetBalance(ctx, feesAccount.GetAddress(), gravityFeeDenom).Return(sdk.NewCoin(gravityFeeDenom, event.Amount))
+
+	expectedEmission := sdk.NewCoin(appParams.BaseCoinUnit, event.Amount)
+	suite.bankKeeper.EXPECT().SendCoinsFromModuleToModule(ctx, gomock.Any(), gomock.Any(), sdk.NewCoins(expectedEmission)).Times(1)
+	suite.auctionKeeper.EXPECT().GetActiveAuctions(ctx).Return([]*auctionTypes.Auction{})
+	suite.bankKeeper.EXPECT().GetBalance(ctx, feesAccount.GetAddress(), gravityFeeDenom).Return(sdk.NewCoin(gravityFeeDenom, event.Amount))
+	suite.auctionKeeper.EXPECT().BeginAuction(ctx, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Times(1)
+
+	require.NotPanics(func() { cellarfeesKeeper.BeginBlocker(ctx) })
+
+	require.Equal(expectedCounters, cellarfeesKeeper.GetFeeAccrualCounters(ctx))
+
+}
+
+func (suite *KeeperTestSuite) TestAuctionBeginWithInSufficientFunds() {
```

Comment:
I don't think I understand what this is testing. Does the module account implicitly have a zero balance? Why is BeginAuction being called if that's the case?